packages/fxa-settings/src/pages/Signup/Confirm/index.tsx

```diff
@@ -32,25 +65,128 @@ const Confirm = ({ email }: RouteComponentProps & ConfirmProps) => {
     instructionText: `Check your email for the confirmation link sent to ${email}`,
   };
 
-  const resendEmailHandler = async () => {
+  // Verifies if we have access to profile info and session token.
+  // If we do, set the email and render the page. If not, we don't know
+  // who the user is and can't confirm if we've sent a confirmation
+  // link, and we can't send a new one without an email address => redirect
+  // the user to retry signup.
+  useEffect(() => {
+    try {
+      if (sessionTokenId) {
+        getEmail();
+      } else {
+        redirectToSignup();
+      }
+    } catch (e) {
+      redirectToSignup();
+    }
+  }, [getEmail, sessionTokenId]);
+
+  useEffect(() => {
+    if (email) {
+      setShowLoadingSpinner(false);
+    }
+  }, [email, setShowLoadingSpinner]);
+
+  const navigateToConfirmCode = useCallback(
+    () =>
+      navigate('/confirm_signup_code?showReactApp=true', {
+        state: { email },
+        replace: true,
+      }),
+    [email, navigate]
+  );
+
+  // TODO implement broker to determine the the next screen
+  const navigateToNextScreen = () => {
+    logViewEvent(viewName, 'verification.success');
+    // TODO check if isForcePasswordChange
+    // navigate('/post_verify/password/force_password_change', {account})
+
+    // default behaviour: '/signup_confirmed'
+    // if Sync: '/connect_another_device'
+    // if Web: '/settings'
+    navigate('/settings', { replace: true });
+  };
+
+  // Adds a timeout before navigating to the /confirm_signup_code page
+  // when the user clicks on the resend email link.
+  useEffect(() => {
+    function onTimeout() {
+      navigateToConfirmCode();
+    }
+    let navigateTimeoutId: NodeJS.Timeout;
+    if (resendStatus === ResendStatus.sent) {
+      navigateTimeoutId = setTimeout(onTimeout, NAVIGATION_TIMEOUT_MS);
+    }
+
+    return () => {
+      clearTimeout(navigateTimeoutId);
+    };
+  }, [navigateToConfirmCode, resendStatus]);
+
+  // Subscribe to account updates to find out if the session becomes verified.
+  // Stay on this page until the session is verified, then automatically
+  // navigate to the next screen.
+  useInterval(async () => {
     try {
-      // TO-DO: signin confirmation email to user.
-      logViewEvent(viewName, 'resend', REACT_ENTRYPOINT);
-      setResendStatus(ResendStatus['sent']);
+      const sessionStatus = await account.isSessionVerified();
+      if (sessionStatus === true) {
+        navigateToNextScreen();
+        setIsPolling(null);
+      } else {
+        return;
```

Comment:
nit: not sure if `else ... return` is necessary since there isn't anything else in this function (except a `catch` that I don't think could be reached if the `account.isSessionVerified()` doesn't `throw` an error anyways).